src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -106,22 +106,24 @@ export const portfolioApi = createApi({
           string,
         ]
         try {
+          if (Math.random() > 0.8) throw new Error('fake failure')
           const adapter = chainAdapters.get(chainId)
           if (!adapter) throw new Error(`no adapter for ${chainId} not available`)
 
           const chainAdaptersAccount = await adapter.getAccount(accountSpecifier)
 
           const portfolioAccounts = { [accountSpecifier]: chainAdaptersAccount }
           const data = accountToPortfolio({ portfolioAccounts, assetIds })
-          // dispatching wallet portfolio, this is done here instead of it being done in onCacheEntryAdded
-          // to prevent edge cases like #820
           dispatch(portfolio.actions.upsertPortfolio(data))
           return { data }
         } catch (e) {
-          const status = 400
-          const data = JSON.stringify(e)
-          const error = { status, data }
-          return { error }
+          const accountId: AccountId = Object.entries(accountSpecifierMap)[0].join(':')
+          moduleLogger.error(e, `error fetching account ${accountId}`)
+          const data = cloneDeep(initialState)
+          data.accounts.ids.push(accountId)
+          data.accounts.byId[accountId] = { assetIds: [] }
+          dispatch(portfolio.actions.upsertPortfolio(data))
```

Comment:
this is our new error state - we tried to load an account and failed. now we upsert an empty account - a valid account will always have at least 1 assetId - the fee asset. an errored account will have a zero length `assetIds`